src/renderer/dx/CustomTextRenderer.cpp

```diff
@@ -531,13 +545,86 @@ using namespace Microsoft::Console::Render;
+
+    // Dig out the box drawing effect parameters.
+    BoxScale scale;
+    RETURN_IF_FAILED(clientDrawingEffect->GetScale(&scale));
+
+    // The scale transform will inflate the entire geometry first.
+    // We want to do this before it moves out of its original location as generally our
+    // algorithms for fitting cells will blow up the glyph to the size it needs to be first and then
+    // nudge it into place with the translations.
+    const auto scaleTransform = D2D1::Matrix3x2F::Scale(scale.HorizontalScale, scale.VerticalScale);
+
+    // Now shift it all the way to where the baseline says it should be.
+    const auto baselineTransform = D2D1::Matrix3x2F::Translation(baselineOrigin.x, baselineOrigin.y);
+
+    // Finally apply the little "nudge" that we may have been directed to align it better with the cell.
+    const auto offsetTransform = D2D1::Matrix3x2F::Translation(scale.HorizontalTranslation, scale.VerticalTranslation);
+
+    // The order is important here. Scale it first, then slide it into place.
+    const auto matrixTransformation = scaleTransform * baselineTransform * offsetTransform;
 
     ::Microsoft::WRL::ComPtr<ID2D1TransformedGeometry> transformedGeometry;
     d2dFactory->CreateTransformedGeometry(pathGeometry.Get(),
-                                          &matrixAlign,
+                                          &matrixTransformation,
                                           transformedGeometry.GetAddressOf());
 
+    // Can be used to see the dimensions after translation.
+    /*D2D1_RECT_F boundsAfter;
+    transformedGeometry->GetBounds(D2D1::IdentityMatrix(), &boundsAfter);*/
+    // Compare this to the original bounds above to see what the matrix did.
+    // To make it useful, first visualize for yourself the pixel dimensions of the cell
+    // based on the baselineOrigin and the exact integer cell width and heights that we're storing.
+    // You'll also probably need the full-pixel ascent and descent because the point we're given
+    // is the baseline, not the top left corner of the cell as we're used to.
+    // Most of these metrics can be found in the initial font creation routines or in
+    // the line spacing applied to the text format (member variables on the renderer).
+    // baselineOrigin = (0, 567)
+    // fullPixelAscent = 39
+    // fullPixelDescent = 9
+    // cell dimensions = 26 x 48 (notice 48 height is 39 + 9 or ascent + descent)
+    // This means that our cell should be the rectangle
+    //
+    //             T=528
+    //           |-------|
+    //       L=0 |       |
+    //           |       |
+    // Baseline->x       |
+    //  Origin   |       | R=26
+    //           |-------|
+    //             B=576
+    //
+    // And we'll want to check that the bounds after transform will fit the glyph nicely inside
+    // this box.
+    // If not? We didn't do the scaling or translation correctly. Oops.
+
+    // Draw the outline then fill it in.

```

Comment:
I'm surprised we draw _and_ fill. Is that the right thing to do? My understanding of font rendering was that we always _filled_ glyph shapes, and outlining them was not correct.


... is outlining them part of making this work??

Reply:
You're looking into that harder than I did. I just had a vague recollection of something looking a little off from a year ago when I did custom rendering so I decided to draw the outline AND fill. +/- I might have seen it in an example. 

I can dig into it and/or rip it and re-run all my tests if you want.

Reply:
Ripped it, it's better.